py/server/deephaven/tree_table.py

```diff
@@ -0,0 +1,110 @@
+
+_JTreeTable = jpy.get_type("io.deephaven.engine.table.hierarchical.TreeTable")
+_JNodeOperationsRecorder = jpy.get_type("io.deephaven.engine.table.hierarchical.TreeTable$NodeOperationsRecorder")
+
+
+class TreeNodeOperationsRecorder(JObjectWrapper, FormatOperationsRecorder,
+                                 SortOperationsRecorder, FilterOperationsRecorder):
+    """Recorder for node-level operations to be applied when gathering snapshots of TreeTable. Supported operations
+    include column formatting, sorting, and filtering.
+
+    Note: It should not be instantiated directly. User code must call :meth:`~TreeTable.node_operation_recorder` to
+    create an instance of the recorder.
+    """
+
+    j_object_type = _JNodeOperationsRecorder
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_node_ops_recorder
+
+    def __init__(self, j_node_ops_recorder: jpy.JType):
+        self.j_node_ops_recorder = j_node_ops_recorder
+
+
+class TreeTable(JObjectWrapper):
+    """ A TreeTable is generated as a result of applying the :meth:`~Table.tree` method on a
+    :class:`~deephaven.table.Table`.
+
+    Note: TreeTable should not be instantiated directly by user code.
+    """
+    j_object_type = _JTreeTable
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_tree_table
+
+    def __init__(self, j_tree_table: jpy.JType, id_col: str, parent_col: str):
+        self.j_tree_table = j_tree_table
+        self.id_col = id_col
+        self.parent_col = parent_col
+
+    def node_operation_recorder(self) -> TreeNodeOperationsRecorder:
+        """Creates a TreepNodeOperationsRecorder for per-node operations to apply during snapshotting of this TreeTable.
+
+        Returns:
+            a TreeNodeOperationsRecorder
+        """
+        return TreeNodeOperationsRecorder(j_node_ops_recorder=self.j_tree_table.makeNodeOperationsRecorder())
+
+    def with_node_operations(self, node_ops: TreeNodeOperationsRecorder) -> TreeTable:
+        """Returns a new TreeTable that will apply the recorded node operations to nodes when gathering snapshots.
+
+        Args:
+            node_ops (TreeNodeOperationsRecorder): the TreeNodeOperationsRecorder containing the node operations to be
+                applied
+
+        Returns:
+            a new TreeTable
+
+        Raises:
+            DHError
+        """
+
+        try:
+            return TreeTable(
+                j_tree_table=self.j_tree_table.withNodeOperations(node_ops.j_node_ops_recorder),
+                id_col=self.id_col, parent_col=self.parent_col)
+        except Exception as e:
+            raise DHError(e, "with_node_operations on TreeTable failed.") from e
+
+    def with_filters(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]) -> TreeTable:
+        """Applies a set of filters to the columns of this TreeTable to produce and return a new TreeTable
```

Comment:
Compare this pydoc to what we have for `where`.  I think `where` is more clear.  https://github.com/deephaven/deephaven-core/blob/main/py/server/deephaven/table.py#L619

Also note that this and the other pydocs in this PR do not match the style of `where`.

Reply:
Changing it to 'where' goes against the wrapping principle of keeping the name as close to the Java one as possible. @ryan your thoughts?

Reply:
I don't think Chip is proposing a rename of the method, just making the docs as good as the method on Table.